libs/s25main/buildings/nobMilitary.h

```diff
@@ -71,6 +71,7 @@ class nobMilitary : public nobBaseMilitary
     bool is_regulating_troops;
     /// Soldatenbesatzung
     OwnedSortedTroops troops;
+    std::array<unsigned, NUM_SOLDIER_RANKS> desired_troops;
```

Comment:
Fixed size required for `sgd.PushContainer(desired_troops);`:

E.g.:
```suggestion
    std::array<uint8_t, NUM_SOLDIER_RANKS> desired_troops;
```